### PR TITLE
Remove python2 from the template

### PR DESCRIPTION
**Pull request by bsipocz**
To address #264

### CODE REVIEW

**Comment by astrofrog** [comment]
Having thought about this a bit more, I'd like to make sure we include https://github.com/astropy/package-template/pull/375 once it's finished before dropping python 2.

**Comment by astrofrog** [comment]
Actually the changes in #375 will require astropy-helpers 3.x, so let's not worry about that.

I'm on board with dropping Python 2 support now, but on the condition that we make a cookiecutter-2.x branch and then include a note in the docs on how to run cookiecutter if you really want to generate a package that works with Python 2. Then we can always fix critical bugs in that branch if needed (but only critical bugs). I think for now we might also want to have the ``cookiecutter-2.x`` branch be rendered to ``master`` rather than do it from the ``cookiecutter`` branch (remember that the master branch is only there for backward compatibility anyway)

**Comment by bsipocz** [comment]
I very much like the idea of making the bugfix branch for possible critical stuff.

However 👎 on making master rendered from that branch. The whole idea of dropping python2 would be to a bit more forcefully nagging packages along the path of dropping it. (As well as I would like using the rendered branch for tagging and then batchpr-ing to packages...). We can have a `rendered-py2` branch if necessary, too, just please not the `master`.

**Comment by astrofrog** [comment]
@bsipocz - ok I think I agree with you, but we have to document this well in the package-template docs.

**Comment by Cadair** [comment]
> As well as I would like using the rendered branch for tagging and then batchpr-ing to packages...

Would probably be better to render for each package, even if you just set "packagename" it would make the diff smaller.

**Comment by bsipocz** [comment]
> Would probably be better to render for each package, even if you just set "packagename" it would make the diff smaller.

No, I'm not willing to go down in that rabbit whole, that'll the the job of the individual maintainers. 
I have the firm opinion to only issue batchPRs to packages for the files that are unambiguous, e.g. no package should modify the content (e.g. _astropy_init.py and alike) and only for tagged package template versions.

**Comment by astrofrog** [comment]
Since we don't have a ``rendered`` branch, I think let's create a ``master-py2`` branch, then in the docs mention that ``master`` is now Python 3-only and that to remain Python 2-compatible, packages should switch to pulling from ``master-py2``. Once that's documented, I think this will be good to go.

**Comment by bsipocz** [comment]
`cookiecutter-2.x` branch has been created as a copy of `cookiecutter` as of today.

**Comment by bsipocz** [comment]
I've added some docs, so this now is ready for final review and merging.

File: {{ cookiecutter.package_name }}/.travis.yml [hunk 1]
```diff
@@ -63,22 +63,17 @@ env:
         # - FLAKE8_OPT='--ignore=E501'
         - FLAKE8_OPT=''
 
+    matrix:
```
Comment (pllim) [inline {{ cookiecutter.package_name }}/.travis.yml]: Why move this up? Just curious.

Comment (bsipocz) [reply {{ cookiecutter.package_name }}/.travis.yml]: some leftover from the rebase. I agree that it shouldn't be up here.

**Comment by bsipocz** [comment]
@pllim - I would leave any `setup.py` related cleanup to #375 or being done separately after that PR if that's all right.

**Comment by astrofrog** [comment]
@bsipocz - can you rebase?

**Comment by bsipocz** [comment]
rebase done

**Comment by bsipocz** [comment]
all green and approved, so I go ahead and merge it.

**Pull request merged**